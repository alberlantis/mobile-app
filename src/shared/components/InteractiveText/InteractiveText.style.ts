import { StyleSheet } from "react-native";

import fonts from "src/theme/fonts";

export default StyleSheet.create({
  container: {
    fontSize: fonts[14],
    fontWeight: "regular",
  },
  interactiveText: {
    textDecorationLine: "underline",
  },
});
