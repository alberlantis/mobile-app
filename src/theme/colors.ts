export default {
  BLACK: "#000",
  BLACK_INPUT: "#28292A",
  WHITE: "#FFF",
  WHITE_BOLD: "rgba(255, 255, 255, 0.66)",
  WHITE_LIGHT: "rgba(255, 255, 255, 0.33)",
  GRAY: "rgba(255, 255, 255, 0.12)",
  ORANGE_PRIMARY_DARK: "#FF7849",
  ORANGE_PRIMARY_LIGHT: "#FF8D3B",
  BLUE_PRIMARY: "#5466D8",
  TRANSPARENT: "transparent",
};
