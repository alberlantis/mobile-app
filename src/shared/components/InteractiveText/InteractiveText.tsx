import React from "react";
import {
  ColorValue,
  Text,
  StyleSheet,
  GestureResponderEvent,
} from "react-native";

import colors from "src/theme/colors";
import s from "./InteractiveText.style";

interface IInteractiveTextProps {
  prefix: string;
  text?: string;
  color?: ColorValue;
  prefixColor?: ColorValue;
  marginBottom?: number;
  onPress(event?: GestureResponderEvent): void;
}

const InteractiveText: React.FC<IInteractiveTextProps> = ({
  prefix,
  prefixColor = colors.WHITE_BOLD,
  text,
  color = colors.WHITE,
  marginBottom,
  onPress,
}) => {
  return (
    <Text
      style={StyleSheet.compose(s.container, {
        color: prefixColor,
        marginBottom,
      })}
      onPress={onPress}
    >
      {`${prefix} `}
      <Text style={StyleSheet.compose(s.interactiveText, { color: color })}>
        {text}
      </Text>
    </Text>
  );
};

export default InteractiveText;
