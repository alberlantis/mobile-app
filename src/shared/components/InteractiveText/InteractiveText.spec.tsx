import { fireEvent, render, screen } from "@testing-library/react-native";

import InteractiveText from "./InteractiveText";
import colors from "src/theme/colors";
import s from "./InteractiveText.style";

const mockOnPress = jest.fn();

describe("InteractiveText", () => {
  describe("general functionality", () => {
    beforeEach(() => {
      render(
        <InteractiveText
          onPress={mockOnPress}
          prefix="prefix-test"
          text="text-test"
        />,
      );
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render interactive text with default white color", () => {
      const prefixAndMain = screen.getByText("prefix-test text-test");
      const mainText = screen.getByText("text-test");
      expect(prefixAndMain).toBeOnTheScreen();
      expect(prefixAndMain).toHaveStyle({
        ...s.container,
        color: colors.WHITE_BOLD,
      });
      expect(mainText).toBeOnTheScreen();
      expect(mainText).toHaveStyle({
        ...s.interactiveText,
        color: colors.WHITE,
      });
    });

    it("should trigger onPress when tapped", () => {
      const mainText = screen.getByText("text-test");
      fireEvent.press(mainText);
      expect(mockOnPress).toHaveBeenCalledTimes(1);
    });
  });

  describe("when color prop is given", () => {
    beforeEach(() => {
      render(
        <InteractiveText
          onPress={mockOnPress}
          prefix="prefix-test"
          text="text-test"
          color={colors.BLACK}
        />,
      );
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render interactive text with given color", () => {
      const prefixAndMain = screen.getByText("prefix-test text-test");
      expect(prefixAndMain).toBeOnTheScreen();
      expect(prefixAndMain).toHaveStyle({
        ...s.container,
        color: colors.WHITE_BOLD,
      });
    });
  });
});
