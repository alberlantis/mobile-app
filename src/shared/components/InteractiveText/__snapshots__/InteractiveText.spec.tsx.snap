// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractiveText general functionality should match snapshot 1`] = `
<Text
  onPress={[MockFunction]}
  style={
    [
      {
        "fontSize": 14,
        "fontWeight": "regular",
      },
      {
        "color": "rgba(255, 255, 255, 0.66)",
        "marginBottom": undefined,
      },
    ]
  }
>
  prefix-test 
  <Text
    style={
      [
        {
          "textDecorationLine": "underline",
        },
        {
          "color": "#FFF",
        },
      ]
    }
  >
    text-test
  </Text>
</Text>
`;

exports[`InteractiveText when color prop is given should match snapshot 1`] = `
<Text
  onPress={
    [MockFunction] {
      "calls": [
        [],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "fontWeight": "regular",
      },
      {
        "color": "rgba(255, 255, 255, 0.66)",
        "marginBottom": undefined,
      },
    ]
  }
>
  prefix-test 
  <Text
    style={
      [
        {
          "textDecorationLine": "underline",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    text-test
  </Text>
</Text>
`;
